server/src/main/java/org/candlepin/pinsetter/tasks/HypervisorUpdateJob.java

```diff
@@ -267,46 +280,42 @@ public void toExecute(JobExecutionContext context) throws JobExecutionException
-                    Now that events no longer serialize whole Objects for the purpose of storing
-                    the oldEntity field, forcing initialization of lazy-loaded collections as a side-effect,
-                    we need to force their initialization before we get to save the result,
-                    to avoid LazyInitializationExceptions when we get to fetch it from the db later.
-                     */
-                    Hibernate.initialize(knownHost.getCapabilities());
-                    Hibernate.initialize(knownHost.getInstalledProducts());
-                    Hibernate.initialize(knownHost.getEntitlements());
-
-                    GuestMigration guestMigration = migrationProvider.get().buildMigrationManifest(incoming,
-                        knownHost);
-
-                    if (consumerResource.performConsumerUpdates(incoming, knownHost, guestMigration,
-                       false)) {
-                        if (guestMigration.isMigrationPending()) {
-                            guestMigration.migrate();
-                        }
-                        else {
-                            consumerCurator.update(knownHost);
-                        }
 
-                        result.updated(updateCheckinTime(knownHost));
+                    GuestMigration guestMigration = new GuestMigration(consumerCurator)
+                        .buildMigrationManifest(incoming, knownHost);
+
+                    boolean factsUpdated = consumerResource.checkForFactsUpdate(knownHost, incoming);
+
+                    if (factsUpdated || guestMigration.isMigrationPending()) {
+                        knownHost.setLastCheckin(new Date());
+                        guestMigration.migrate(false);
+                        result.updated(knownHost);
                     }
                     else {
-                        result.unchanged(updateCheckinTime(knownHost));
+                        result.unchanged(knownHost);
                     }
                 }
                 // update reporter id if it changed
                 if (jobReporterId != null && reportedOnConsumer != null &&
                     reportedOnConsumer.getHypervisorId() != null &&
                     (reportedOnConsumer.getHypervisorId().getReporterId() == null ||
-                    !jobReporterId.contentEquals(reportedOnConsumer.getHypervisorId().getReporterId()))) {
+                        !jobReporterId.contentEquals(reportedOnConsumer.getHypervisorId().getReporterId()))) {
                     reportedOnConsumer.getHypervisorId().setReporterId(jobReporterId);
                 }
                 else if (jobReporterId == null) {
                     log.debug("hypervisor checkin reported asynchronously without reporter id " +
                         "for hypervisor:{} of owner:{}", hypervisorId, ownerKey);
                 }
             }
+
+            for (Consumer consumer : hypervisorConsumersMap.getConsumers()) {
```

Comment:
I think we need to set the lastUpdatedTime or lastCheckinTIme or something here. What'd we end up doing in the 2.2 version of this change?

Reply:
We did update the lastCheckinTime here in the 2.2 version. I've set the lastCheckinTime during createConsumerForHypervisorId in this branch, to avoid multiple update calls generated by hibernate (we do one update in GuestMigration and a create for new consumers at the end of the job).